js/modules/k6/experimental/tracing/trace_id.go

```diff
@@ -45,6 +46,23 @@ type TraceID struct {
 	// When encoded, it should be in a format occupying the last 8 bytes of
 	// the trace-id, and should ideally be encoded as nanoseconds.
 	Time time.Time
+
+	// randSource holds the randomness source to use when encoding the
+	// trace-id. The `rand.Reader` should be your default pick. But
+	// you can replace it with a different source for testing purposes.
+	randSource io.Reader
+}
+
+// NewTraceID produces a new TraceID with the given prefix, code and time.
+//
+// It sets the traceID randomness source to the `rand.Reader` as a default.
+func NewTraceID(prefix int16, code int8, t time.Time, randSource io.Reader) TraceID {
+	return TraceID{
+		Prefix:     prefix,
+		Code:       code,
+		Time:       t,
+		randSource: rand.Reader,
```

Comment:
off-topic: Does the source need to be crypto secure?

Reply:
As far as I know, it doesn't need to. I am switching to `math/rand.Rand` 👍🏻